lib/streamlit/elements/utils.py

```diff
@@ -59,7 +54,9 @@ def check_session_state_rules(
         return
 
     if not writes_allowed:
-        raise StreamlitAPIException(SESSION_STATE_WRITES_NOT_ALLOWED_ERROR_TEXT)
+        raise StreamlitAPIException(
+            f'Values of the widget with key "{key}" cannot be set using `st.session_state`.'
```

Comment:
nit: "Values of the..." -> "Values for ..."

Reply:
Updated 👍